cc_macro.c

```diff
@@ -747,6 +862,27 @@ void eat_block(void)
 	if(match("\n", macro_token->prev->s)) macro_token = macro_token->prev;
 }
 
+struct token_list* deep_copy_token_list(struct token_list* from)
+{
+	if(from == NULL)
+	{
+		return NULL;
+	}
+
+	struct token_list* to = calloc(1, sizeof(struct token_list));
+
+	to->next = deep_copy_token_list(from->next);
+	to->locals = from->locals;
+	to->prev = from->prev;
```

Comment:
I checked again and it doesn't seem like there's actually a problem with `t->prev->next` not being `token` anywhere outside of macro handling so it might just be me messing up the copy here.

After this gets merged and I have more time I'll work on multi token arguments.